tests/ui/asm/cfg-parse-error.rs

```diff
@@ -0,0 +1,45 @@
+//@ needs-asm-support
+#![feature(asm_cfg)]
+
+use std::arch::asm;
+
+fn main() {
+    unsafe {
+        // Templates are not allowed after operands (even if the operands are configured out).
```

Comment:
Is this a reasonable restriction?  I'm thinking here about macros that might want to expand something like:

```rust
cfg_match! { 
    one => { "jmp here", options(some_option), };
    two => { "jmp there", options(other_option), };
}
```

Into:

```rust
#[cfg(one)]
"jmp here",
#[cfg(one)]
options(some_option),
#[cfg(and(not(one), two)]
"jmp there",
#[cfg(and(not(one), two)]
options(other_option),
```

Reply:
Hmm yeah that's a good point.

We'd have to make some further parser changes to make that work. Currently the parsing rules change based on whether a template has already been parsed (mostly for error message quality).

A second issue is that templates are currently parsed with the general expression parser, but some of the non-template arguments parse as valid expressions (e.g. `options(noreturn)` looks like a function call, and `const { 5 }` is also a valid expression).

A solution is to instead parse specifically string literals and macro calls, those are templates, and if those fail to parse, try to parse an operand.